File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/utils/ConnectionDetector.java
package com.c_store.zhiyazhang.cstoremanagement.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.widget.Toast;

import com.c_store.zhiyazhang.cstoremanagement.R;

/**
 * Created by zhiya.zhang
 * on 2017/5/19 10:29.
 * 检测网络工具类
 */

public class ConnectionDetector {
    private volatile static ConnectionDetector connectionDetector;
    private ConnectionDetector(){}
    public static ConnectionDetector getConnectionDetector(){
        if (connectionDetector==null){
            synchronized (ConnectionDetector.class){
                if (connectionDetector==null){
                    connectionDetector=new ConnectionDetector();
                }
            }
        }
        return connectionDetector;
    }

    private static final Context context=MyApplication.getContext();

    public static boolean isOnline() {
        ConnectivityManager connectivity = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivity != null) {
            NetworkInfo info = connectivity.getActiveNetworkInfo();
            if (info != null && info.isConnected()) {
                // 当前网络是连接的
                if (info.getState() == NetworkInfo.State.CONNECTED) {
                    // 当前所连接的网络可用
                    return true;
                }
            }
        }
        Toast.makeText(context,
                context.getResources().getString(R.string.noItent),
                Toast.LENGTH_LONG)
                .show();
        return false;
    }
}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/utils/ConnectionDetector.kt
package com.cstore.zhiyazhang.cstoremanagement.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import com.cstore.zhiyazhang.cstoremanagement.R
import com.zhiyazhang.mykotlinapplication.utils.MyApplication

/**
 * Created by zhiya.zhang
 * on 2017/5/19 10:29.
 * 检测网络工具类
 */
class ConnectionDetector private constructor() {

    companion object {
        fun getConnectionDetector(): ConnectionDetector {
            return Inner.cd
        }
        private val context = MyApplication.instance().applicationContext
    }

    private object Inner {
        val cd = ConnectionDetector()
    }


    /**
     * 检测是否有网络
     * @return 是\否
     */
    val isOnline: Boolean
        get() {
            val connectivity = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val info = connectivity.activeNetworkInfo
            if (info.isConnected) if (info.state == NetworkInfo.State.CONNECTED) return true
            MyToast.getShortToast(context.getString(R.string.noInternet))
            return false
        }
}
